lib/datadog/tracing/trace_operation.rb

```diff
@@ -450,7 +450,9 @@ def build_trace(spans, partial = false)
           service: service,
           tags: meta,
           metrics: metrics,
-          root_span_id: !partial ? root_span && root_span.id : nil
+          root_span_id: !partial ? root_span && root_span.id : nil,
+          profiling_enabled:
+            !!(defined?(Datadog::Profiling) && Datadog::Profiling.respond_to?(:enabled?) && Datadog::Profiling.enabled?), # rubocop:disable Style/DoubleNegation
```

Comment:
I think this should be checked earlier and cached carefully if possible.

In today's world, I believe we can move this into the `Tracing::Tracer` object initialization process with no loss of correctness, then pass the value here.

Reply:
Good point. Fixed in cbe0317323587b27a542255211f4b5636a36ff56.

I had to do it lazily instead of during initialization to avoid any potential of loops (Profiling::Component taking an optional_tracer and doing something similar in the other direction), but otherwise it was a straightforward change.